### PR TITLE
Markdown [WIP]

### PR DESCRIPTION
**Pull request by johnxnguyen**

### CODE REVIEW

**Comment by JPonte** [comment]
build again

**Comment by deanrobertcook** [comment]
build again

File: app/src/main/java/com/waz/zclient/markdown/MarkdownTextview.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import com.waz.zclient.markdown.spans.GroupSpan;
+import com.waz.zclient.markdown.spans.commonmark.ImageSpan;
+import com.waz.zclient.markdown.spans.commonmark.LinkSpan;
+import com.waz.zclient.ui.text.TypefaceTextView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MarkdownTextview extends TypefaceTextView {
```
Comment (JPonte) [inline app/src/main/java/com/waz/zclient/markdown/MarkdownTextview.java]: I guess the V could be capital to be consistent, but not important.

File: app/src/main/java/com/waz/zclient/markdown/visitors/SpanRenderer.kt [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+        writer.set(prefixSpan, prefixStart, writer.cursor)
+
+        // write the content
+        writer.tabIfNeeded()
+        visitChildren(listItem)
+        writeLineIfNeeded(listItem)
+
+        val start = writer.retrieveCursor()
+
+        // TODO: General clean up here
```
Comment (JPonte) [inline app/src/main/java/com/waz/zclient/markdown/visitors/SpanRenderer.kt]: There are a few TODOs in this file. I imagine the image one should be a future improvement, but what about the others?

File: app/src/main/java/com/waz/zclient/markdown/MarkdownTextview.java (hunk 1 above)
Comment (johnxnguyen) [reply app/src/main/java/com/waz/zclient/markdown/MarkdownTextview.java]: Good catch, was a typo

**Comment by someoneEsle** [comment]
build again

**Comment by someoneEsle** [comment]
- [ ] Question 1
Should the following string be in italics?
*line1
line2
line3*

**Comment by someoneEsle** [comment]
- [ ] Bug 1
Markdown is not rendered in notifications

**Comment by johnxnguyen** [comment]
@someoneEsle 

**Q1:** No, because line breaks create new paragraphs and bold and italics do not render over paragraphs. 

**B1:** Sorry I forgot to mention this in the description. I was planning to implement this in a future iteration, but if it something that the Android team feels would be nice to have with the initial release, then we can arrange that.

**Pull request merged**